hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadIncrementalRelation.scala

```diff
@@ -209,7 +209,14 @@ trait HoodieIncrementalRelationTrait extends HoodieBaseRelation {
 
   protected lazy val includedCommits: immutable.Seq[HoodieInstant] = queryContext.getInstants.asScala.toList
 
-  protected lazy val commitsMetadata = includedCommits.map(getCommitMetadata(_, super.timeline)).asJava
+  protected lazy val commitsMetadata = includedCommits.map(
+    i => {
```

Comment:
Let's check why we need to resolve the commit metadata for an archived instant, even in incremental query scenarios, the reader should fallback to snapshot query with specific filters instead of resolving the data files from commit metadata (because the data files are very probably been cleaned.)

Reply:
I have checked the usage of this `commitsMetadata`, it decodes the data files and use a `fs.exists` check for the data files as a part of the fallback strategy. So the fix should be reasonable.

Reply:
Created HUDI-8624 to revisit why we need commit metadata from timeline history. For now looks like we need this logic. If full table scan is enabled, `commitsMetadata` is not loaded since it's lazy.